TelemetrySite/server/data_upload_scripts/data_upload_api.py

```diff
@@ -0,0 +1,92 @@
+from flask.views import MethodView
+from flask import request, jsonify
+from werkzeug.utils import secure_filename
+import os
+from data_upload_scripts.data_upload import file_convert, get_progress
+from utils import exec_get_one, exec_get_all
+
+
+class data_upload_api(MethodView):
+
+    ALLOWED_EXTENSIONS = {"mf4"}
+    UPLOAD_FOLDER = os.path.dirname(__file__) + "/data_upload_file"
+
+    def __init__(self):
+        # Create upload folder if it doesn't exist
+        if not os.path.exists(self.UPLOAD_FOLDER):
+            os.makedirs(self.UPLOAD_FOLDER)
+
+    ## Get the current progress of the
+    #  current data upload action
+    #  TODO revisit this. Not quite happy with where this call is
+    def get(self, contextId):
+
+        return jsonify({"progress": get_progress(contextId)}), 200
+
+    def post(self, contextId):
+        # Check if the post request has the file part
+
+        if "file" not in request.files:
+            return jsonify({"error": "No file uploaded"}), 400
+        elif "contextID" not in request.form:
+            return jsonify({"error": "No id passed"}), 400
+
+        config_id = request.form["contextID"]
+        file = request.files["file"]
+
+        # check for a duplicate context id value
+        sql_check_unique = "SELECT 1 FROM canMessage WHERE contextId = %s"
+        sql_check_result = exec_get_one(
+            sql_check_unique,
+            [config_id],
+        )
+
+        if not sql_check_result is None:
+            return jsonify({"error": "Attempt to upload duplicate context id"}), 400
+
+        # check for key in context db
+        sql_check_exists = "SELECT 1 FROM context WHERE id = %s"
+        sql_check_result = exec_get_one(
+            sql_check_exists,
+            [config_id],
+        )
+        # return an error if id doesn't exists
+        if sql_check_result is None:
+            return (
+                jsonify({"error": "Passed context id does not exits in database"}),
+                400,
+            )
+
+        # ensure the file actually contains a valid file name
+        if not file or file.name == "":
+            return jsonify({"error": "No file uploaded"}), 400
+
+        if self.file_type_check(file.filename):
+            # Secure name for best practice
```

Comment:
Change this comment to make it clearer what it actually means to "secure" a filename

Reply:
Updated comment to describe what secure_filename does and what it changes in the name